haystack/nodes/other/route_documents.py

```diff
@@ -38,43 +51,89 @@ def __init__(self, split_by: str = "content_type", metadata_values: Optional[Lis
 
         self.split_by = split_by
         self.metadata_values = metadata_values
+        self.return_remaining = return_remaining
+
+        if self.split_by != "content_type" and not isinstance(self.metadata_values, list):
```

Comment:
Can't we just check that `self.metadata_values` is not `None` in here?
The `__init__` type hint already implies that it must be a `List`.

Reply:
Also what if `self.metadata_values` is an empty list?

Reply:
Good points, I'll make adjustments. This was mostly a copy from existing code.